mava/logger.py

```diff
@@ -13,119 +13,120 @@
-        else:
-            prefix = "evaluator/"
-            episodes_info = metrics.episodes_info
-
-        # Flatten metrics info.
+    ) -> Tuple[str, float]:
+        """Log the environment's episodes metrics."""
         episodes_return = jnp.ravel(episodes_info["episode_return"])
         episodes_length = jnp.ravel(episodes_info["episode_length"])
-        steps_per_second = episodes_info["steps_per_second"]
 
-        # Log metrics.
-        if logger.should_log:
-            logger.log_stat(
-                f"{prefix}mean_episode_returns", float(np.mean(episodes_return)), t_env, eval_step
+        if self.logger.should_log:
+            self.logger.log_stat(
+                f"{prefix}/mean_episode_returns", float(jnp.mean(episodes_return)), t_env, eval_step
             )
-            logger.log_stat(
-                f"{prefix}mean_episode_length", float(np.mean(episodes_length)), t_env, eval_step
+            self.logger.log_stat(
+                f"{prefix}/mean_episode_length", float(jnp.mean(episodes_length)), t_env, eval_step
             )
-            logger.log_stat(f"{prefix}steps_per_second", steps_per_second, t_env, eval_step)
-
-            if trainer_metric:
-                logger.log_stat(f"{prefix}total_loss", float(np.mean(total_loss)), t_env)
-                logger.log_stat(f"{prefix}value_loss", float(np.mean(value_loss)), t_env)
-                logger.log_stat(f"{prefix}loss_actor", float(np.mean(loss_actor)), t_env)
-                logger.log_stat(f"{prefix}entropy", float(np.mean(entropy)), t_env)
 
         log_string = (
             f"Timesteps {t_env:07d} | "
-            f"Mean Episode Return {float(np.mean(episodes_return)):.3f} | "
-            f"Std Episode Return {float(np.std(episodes_return)):.3f} | "
-            f"Max Episode Return {float(np.max(episodes_return)):.3f} | "
-            f"Mean Episode Length {float(np.mean(episodes_length)):.3f} | "
-            f"Std Episode Length {float(np.std(episodes_length)):.3f} | "
-            f"Max Episode Length {float(np.max(episodes_length)):.3f} | "
-            f"Steps Per Second {steps_per_second:.2e} "
+            f"Mean Episode Return {float(jnp.mean(episodes_return)):.3f} | "
+            f"Std Episode Return {float(jnp.std(episodes_return)):.3f} | "
+            f"Max Episode Return {float(jnp.max(episodes_return)):.3f} | "
+            f"Mean Episode Length {float(jnp.mean(episodes_length)):.3f} | "
+            f"Std Episode Length {float(jnp.std(episodes_length)):.3f} | "
+            f"Max Episode Length {float(jnp.max(episodes_length)):.3f} | "
         )
 
+        if "steps_per_second" in episodes_info.keys():
+            steps_per_second = episodes_info["steps_per_second"]
+            if self.logger.should_log:
+                self.logger.log_stat(
+                    f"{prefix}steps_per_second", steps_per_second, t_env, eval_step
+                )
+            log_string += f"Steps Per Second {steps_per_second:.2e}"
```

Comment:
oh this is in the sebulba arch :sweat_smile: (I made this logger to adapt it to sebulba too) shall I remove it for now the if statement

Reply:
No I think keep it, but we should always include it in the `episode_info` (even in anakin) otherwise we're going to have a whole lot of `if` statements to check what we have in the dict vs what we don't have for different systems.